docs/data/date-pickers/custom-field/BrowserV7Field.tsx

```diff
@@ -101,18 +104,9 @@ const BrowserTextField = React.forwardRef(
 
 const BrowserDateField = React.forwardRef(
   (props: DatePickerFieldProps, ref: React.Ref<HTMLDivElement>) => {
-    const { slots, slotProps, ...textFieldProps } = props;
-
-    const fieldResponse = useDateField<true, typeof textFieldProps>(textFieldProps);
-
-    /* If you don't need a clear button, you can skip the use of this hook */
-    const processedFieldProps = useClearableField({
```

Comment:
I want to migrate away from this hook in favor of a single built-in component that applies all the adornment (the one to clean the value and the one to open the picker).

And while working on the new component, I came to the conclusion that it was weird to have a hook that adds `@mui/material` stuff and to expose it to be used in Joy UI and Browser demos for which the goal is to **not** use `@mui/material`.

I think it's better to just not use the hook and add a comment for the `clearable` and `onClear` props so that people can build their own UIs.
Also note that once we have the new Base UI DX this will be a lot easier to do since we will just have an unstyled component the clears the value.

Reply:
makes perfect sense